subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaPlugin.java

```diff
@@ -430,12 +430,8 @@ public BuildableJavaComponentImpl(JavaPluginConvention convention) {
             this.convention = convention;
         }
 
-        public Collection<String> getRebuildTasks() {
-            return Arrays.asList(BasePlugin.CLEAN_TASK_NAME, JavaBasePlugin.BUILD_TASK_NAME);
-        }
-
         public Collection<String> getBuildTasks() {
-            return Arrays.asList(JavaBasePlugin.BUILD_TASK_NAME);
+            return Collections.singletonList(JavaBasePlugin.BUILD_TASK_NAME);
```

Comment:
Nit-pick: could use `Collections.singleton()` instead, since `List` semantics adds nothing here.

Reply:
`Collections.singleton()` it is.